### PR TITLE
icu4c 72.1

### PR DESCRIPTION
**Pull request by Bo98**
It's icu4c time.

---

Created with `brew bump-formula-pr`.

---

TODO list:
- [x] CODEOWNERS to avoid conflicts - #120177
- [x] Fix `clickhouse-odbc` test failure - #120164
- [x] Fix `mpd` build failure - #120259
- [x] Fix `jena` link failure - #120261
- [x] Fix `corepack` link failure - #120357
- [x] Fix `puzzles` link failure - #120358
- [x] Fix `page` link failure - #120359
- [x] Revision bump `simdutf` & `qt@5` / Defer `planck`
- [x] Identify and delete bad uploads from previous run. EDIT: Done for most formulae. Remaining are:
   - [x] `boost@1.76`
   - [x] `node@16`
   - [x] `postgresql@13`
   - [x] `texlive`
   - [x] `tracker`
   - [x] `zk`
- [x] Run and merge via large runner (or local upload if any failure).
- [x] Revert CODEOWNERS - #120681
- [x] Follow up on failures
   - [x] `swift` build failure (Linux) - https://github.com/Homebrew/homebrew-core/actions/runs/3925732531
   - [x] `webkitgtk` & `planck` revision bump - #120680
   - [x] `tin` revision bump and explicit `icu4c` dep (or drop `icu4c` linkage) - #120682
   - [x] `unar` revision bump and explicit `icu4c` dep (or drop `icu4c` linkage) - #120684

### CODE REVIEW

**Comment by Bo98** [comment]
Got revision bumps ready to go. It's the usual plus a few extra Linux-only ones now tha libxml2 pulls in icu4c:

* dcmtk (used for: https://github.com/DCMTK/dcmtk/blob/31ae87d57edf3f5a441cae64869c7337b29213b2/INSTALL#L296-L375)
* dovecot (allows full text search to lowercase UTF8 characters or normalize text - https://doc.dovecot.org/settings/plugin/fts-plugin/)
* openttd (https://wiki.openttd.org/en/Archive/Compilation%20and%20Ports/Cross-compiling%20for%20Windows#icu)
* samba (UTF-8 normalisation - https://github.com/samba-team/samba/commit/107020793c7ea44e5e776a1401bbf4f8ccb9bd85)

Listing these in case they are also desired on macOS (or not at all).

**Comment by iMichka** [comment]
I'm ok with that, let's ship the revision bumps separately, else this PR will be stuck for days.

**Review by iMichka — DISMISSED** [review]

**Comment by Bo98** [comment]
There's a lot of important formulae so I don't think every single one should be separate.

Just haven't pushed yet because I was waiting until after the CI maintenance.

**Comment by SMillerDev** [comment]
This really can't be closed by stalebot.

**Comment by cho-m** [comment]
Cancelling remaining Linux run as the self-hosted has some stale data and didn't build `icu4c`. Probably need to be recreate runner.

**Comment by cho-m** [comment]
Will need a rebase for `qt` and `php`. I'm guessing new `php@8.1` will need revision bump too.

**Comment by SMillerDev** [comment]
Got the revbumps for ARM. Pushing again directly.

**Comment by fxcoudert** [comment]
There is already a conflict for `zebra`. Should we skip recursive dependents? Or skip dependents altogether? Otherwise I'm not seeing us merge this anytime soon…

**Comment by chenrui333** [comment]
> There is already a conflict for `zebra`. Should we skip recursive dependents? Or skip dependents altogether? Otherwise I'm not seeing us merge this anytime soon…

yeah, we can have this run done and then skip the dependants to get it merged.

**Comment by cho-m** [comment]
macOS:
- Intel - `chakra` linkage failure to icu4c
- Monterey - `swift` build failed with stack dump
- ARM Big Sur - `qt@5` audit failed with shim reference
- Intel Big Sur - taking a bit too long to build. Over 2 days. Not sure if still running or stuck. Also, given that dependent tests took 12-14 hours on other Intel runners, may time out.

Linux:
- `qt` still needs test fix
- `libmspub`, `postgis`, `swift`, and `zorba` all failed to build
- Linkage failures that need to be identified and added to follow up PR(s). Could add revision bumps here, but will make it harder to finish PR.

**Comment by chenrui333** [comment]
> * Linkage failures that need to be identified and added to follow up PR(s). Could add revision bumps here, but will make it harder to finish PR.

yeah, I wonder they should be addressed in the subsequent batch PR updates.

**Comment by chenrui333** [comment]
previous run, https://github.com/Homebrew/homebrew-core/actions/runs/3712606085

**Comment by cho-m** [comment]
May want to remove `swift` and handle in separate PR or disable on Intel Big Sur as it is reason why runner got stuck (about 1 day waiting for `swift` build):
```
Sat, 17 Dec 2022 23:07:26 GMT ==> brew install --verbose --build-bottle swift
Sun, 18 Dec 2022 21:10:45 GMT Error: The operation was canceled.
```

**Comment by SMillerDev** [comment]
I'm not sure how to do the codeowners. Could you do it? Because I'm definitely planning on merging this ASAP

**Comment by p-linnane** [comment]
CODEOWNERS PR: https://github.com/Homebrew/homebrew-core/pull/118540

**Comment by cho-m** [comment]
It is possible that Intel Big Sur is stuck on `swift` again, but hard to tell as that runner is also the slowest one. It has 17 hours left before hitting the 3 day limit.

**Comment by SMillerDev** [comment]
Once intel Big Sur finishes I'm going to merge this.

**Comment by p-linnane** [comment]
@cho-m was right about it getting stuck on Swift again. Bottles are uploading now.

**Comment by SMillerDev** [comment]
> Error: The log was not found. It may have been deleted based on retention settings.

Maybe just trigger a merge and rebuild bottles after? Usage for Big Sur is only 11% anyway, and I doubt a rebuild will help.

**Comment by SMillerDev** [comment]
retriggered on a large runner

**Comment by SMillerDev** [comment]
https://github.com/Homebrew/homebrew-core/actions/runs/3753212891/jobs/6376250226

**Comment by cho-m** [comment]
Just copying summary of failures that we may need to follow up on:

ARM Ventura:
```
Error: 2 failed steps!
brew fetch --retry tarantool --build-bottle --force
brew install --verbose --build-bottle tarantool
```

ARM/Intel Monterey:
```
Error: 1 failed step!
brew install --verbose --build-bottle swift
```

ARM Big Sur:
```
Error: 1 failed step!
brew install --verbose --build-bottle php@8.1
```

Linux:
```
Error: 4 failed steps!
brew install --verbose --build-bottle libmspub
brew install --verbose --build-bottle postgis
brew install --verbose --build-bottle swift
brew install --verbose --build-bottle zorba

Error: 22 failed steps!
brew install --only-dependencies osm2pgrouting
brew install osm2pgrouting
brew install --only-dependencies pgrouting
brew install pgrouting
brew test --retry --verbose pktanon
brew test --retry --verbose shibboleth-sp
brew linkage --test webkitgtk
brew test --retry --verbose webkitgtk
brew linkage --test gnustep-base
brew test --retry --verbose gnustep-base
brew linkage --test planck
brew test --retry --verbose planck
brew test --retry --verbose bazarr
brew linkage --test dcmtk
brew test --retry --verbose dcmtk
brew test --retry --verbose gdal
brew install --only-dependencies spatialite-gui
brew install spatialite-gui
brew test --retry --verbose streamlink
brew install --only-dependencies virtualpg
brew install virtualpg
brew test --retry --verbose grantlee
```

**Comment by p-linnane** [comment]
Looks like it's not happy about the ones that have missing bottles.
```
/usr/bin/skopeo inspect --raw docker://ghcr.io/homebrew/core/libcdr:0.1.7_4 --creds=brewtestbot:******
==> Uploading libcdr--0.1.7_4.monterey.bottle.tar.gz
==> Uploading libcdr--0.1.7_4.arm64_ventura.bottle.tar.gz
==> Uploading libcdr--0.1.7_4.arm64_monterey.bottle.tar.gz
==> Uploading libcdr--0.1.7_4.ventura.bottle.tar.gz
==> Uploading libcdr--0.1.7_4.big_sur.bottle.tar.gz
Error: No such file or directory @ rb_sysopen - libcdr--0.1.7_4.big_sur.bottle.tar.gz
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/digest.rb:50:in `initialize'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/digest.rb:50:in `open'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/digest.rb:50:in `file'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/digest.rb:35:in `file'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:429:in `write_tar_gz'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:315:in `block in upload_bottle'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:300:in `each'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:300:in `map'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:300:in `upload_bottle'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:62:in `block in upload_bottles'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:61:in `each'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:61:in `upload_bottles'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dev-cmd/pr-upload.rb:160:in `pr_upload'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:97:in `<main>'
Error: Failure while executing; `/home/linuxbrew/.linuxbrew/bin/brew pr-upload --debug --committer=BrewTestBot\ \<1589480\+BrewTestBot@users.noreply.github.com\> --root-url=https://ghcr.io/v2/homebrew/core` exited with 1.
```

**Comment by cho-m** [comment]
Since we are trying for re-run (which may take some time as single self-hosted runner is queued up for a few days), we may as well try to fix up all other failures.

Linux build failures:
- [ ] `brew install --verbose --build-bottle libmspub` - #118828
- [ ] `brew install --verbose --build-bottle postgis`
- [ ] `brew install --verbose --build-bottle zorba`

---

On macOS, I wonder if we should consider making `clickhouse-odbc` also use `unixodbc`. Not sure on difference with `libiodbc`, but we currently have more formulae that use `unixodbc` (like `php` and `erlang`), so `clickhouse-odbc` probably has higher chance of conflict right now. Other than this formula, we only use `libiodbc` for the `qt-libiodbc` variant.

Alternatively, could make `libiodbc` keg-only, but would need to make sure it doesn't cause issues when both are installed.

**Comment by cho-m** [comment]
Dropping `no long build conflict` as we should merge other PRs now and rebase once `swift` is ready (#114549).

**Comment by cho-m** [comment]
We probably shouldn't run this until `swift` is fixed as we will hit the same issue on Big Sur.

**Comment by SMillerDev** [comment]
Dropping it seems fine to me.

**Comment by SMillerDev** [comment]
cancelling the rest of the run because swift got some movement again and tesseract needs a bump.
```
  ==> Installing clickhouse-odbc dependency: libiodbc
  Error: Cannot install libiodbc because conflicting formulae are installed.
    unixodbc: because both install `odbcinst.h`
```

**Comment by cho-m** [comment]
- [ ] `manticoresearch` failed with checksum mismatch
- [ ] `texlive` failed on missing `pkg-config`. Trying to fix in #119768

**Comment by cho-m** [comment]
Rebased and added some addition revision bumps for Linux failures excluding `webkitgtk` (Linux-only so a failure would make it require manually dropping commit to merge)

However not ready to run yet.
1. Need a fix for `manticoresearch` first, e.g. #119811
2. May want to try to improvde `clickhouse-odbc`. Options:
   - Switch to `unixodbc`. I think this was considered in original PR but not done as `libiodbc` was the standard option for macOS.
   - Consider making `libiodbc` keg-only. Right now, `unixodbc` is more popular as it is a dependency for PHP/Erlang/etc

**Comment by cho-m** [comment]
> > Switch to unixodbc. I think this was considered in original PR but not done as libiodbc was the standard option for macOS.
> 
> I think this is the best option in light of the lacking audit.

Just remembered that we previously had issues with `brew upgrade` when switching dependency for conflicting formulae. May still need to make one keg-only.

**Comment by cho-m** [comment]
Updating CODEOWNERS for new round in #120177. If anyone else wants to be added, can comment there.

Will update first post with TODO items to avoid getting lost as we rebase.

**Comment by cho-m** [comment]
`mpd` failed on Monterey and later, but was able to build a bottle on Big Sur so can still merge if rest of results are okay. Opened PR #120259 to look into fix.

`php` failed on ARM Big Sur. Looks like Apple Clang crash. Can try bottle job afterward.

Also need someone to help remove some uploads from previous attempt for `boost@1.76`, `node@16`, `postgresql@13`, `texlive`, `tracker`, `zk`. I was able to fix the other ~30 formulae but don't have permissions to fix these.

---

Adding these tasks to checklist on first post.

File: Formula/planck.rb
Comment (cho-m) [inline Formula/planck.rb:7]: If we need another run, will defer this to later as it will probably fail on Linux given `webkitgtk` dependency
```suggestion
```

**Comment by cho-m** [comment]
12-arm64:
```
Error: 3 failed steps!
brew install jena
brew linkage --test simdutf
brew test --retry --verbose simdutf
```

May fix these and restart to try to get a clean run on some of our runners. Right now, all are going to fail.

**Comment by cho-m** [comment]
`planck` failed on Linux as I expected. Will defer to separate PR as I don't want to risk `webkitgtk` failing if I revision bump here.

`swift` also failed but I can't see reason why. May need to wait for run to end as it isn't showing up in raw logs.

**Comment by cho-m** [comment]
@Bo98 `swift` failure on Linux:
```
FAILED: lib/libswiftDemangle.so 
: && /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/linux/super/llvm_clang++ -fPIC -Wno-unknown-warning-option -Werror=unguarded-availability-new -fno-stack-protector -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-class-memaccess -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -fdiagnostics-color -ffunction-sections -fdata-sections -Werror=switch -Wdocumentation -Wimplicit-fallthrough -Wunreachable-code -Woverloaded-virtual -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -O3 -DNDEBUG  -target x86_64-unknown-linux-gnu -fuse-ld=gold -shared -Wl,-soname,libswiftDemangle.so -o lib/libswiftDemangle.so lib/SwiftDemangle/CMakeFiles/swiftDemangle.dir/SwiftDemangle.cpp.o lib/SwiftDemangle/CMakeFiles/swiftDemangle.dir/MangleHack.cpp.o -L/tmp/swift-20230110-2076873-jfvu2m/build/llvm-linux-x86_64/./lib -Wl,-rpath,"\$ORIGIN:\$ORIGIN/../lib:\$ORIGIN/../lib/swift/linux"  lib/libswiftDemangling.a && :
/home/linuxbrew/.linuxbrew/opt/binutils/bin/ld.gold: error: cannot open /home/linuxbrew/.linuxbrew/lib/libgcc_s.so.1: No such file or directory
/home/linuxbrew/.linuxbrew/opt/binutils/bin/ld.gold: error: cannot open /home/linuxbrew/.linuxbrew/lib/libgcc_s.so.1: No such file or directory
clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by cho-m** [comment]
Linux notes:
- Going to add `qt@5` for next run as it caused a lot of dependent failures.
- Probably going to ignore `swift` issue. May be from `gcc` or something else in build environment. Can try to bottle later if it fails again.
- `tin` and `unar` are linked on Linux. Added to checklist to follow up on later
- `qt-percona-server` had odd failure as it's RPATH is missing `percona-server` and was linked via `HOMEBREW_PREFIX/lib`. However, this is not related to PR so can ignore.


Once #120360 is in, I'll restart the run.

**Comment by cho-m** [comment]
- ARM all passed. 
- Linux failures are expected and can be followed up later.
- Waiting on Intel macOS results.

Old uploaded bottles still need to be removed if we want to try to use workflow.

**Review by cho-m — DISMISSED** [review]
36GB of bottles. Hopefully self-hosted runner is sufficient. Will trigger a merge with large runner option, but also downloading a local copy in case something goes wrong.

**Comment by chenrui333** [comment]
🤞

**Comment by p-linnane** [comment]
Ran out of disk space
```
cp: error writing '/tmp/11351220230114-2350-z1ry6/freeling--4.2_11.arm64_ventura.bottle.tar.gz': No space left on device
Error: Failure while executing; `cp -pR /tmp/d20230114-2350-70d9b8/freeling--4.2_11.arm64_ventura.bottle.tar.gz /tmp/11351220230114-2350-z1ry6/freeling--4.2_11.arm64_ventura.bottle.tar.gz` exited with 1. Here's the output:
cp: error writing '/tmp/11351220230114-2350-z1ry6/freeling--4.2_11.arm64_ventura.bottle.tar.gz': No space left on device
```

**Comment by Bo98** [comment]
I can increase the disk space of the runner. Surprised it hit it though.

**Comment by Bo98** [comment]
> @Bo98 `swift` failure on Linux:
> 
> ```
> FAILED: lib/libswiftDemangle.so 
> : && /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/linux/super/llvm_clang++ -fPIC -Wno-unknown-warning-option -Werror=unguarded-availability-new -fno-stack-protector -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-class-memaccess -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -fdiagnostics-color -ffunction-sections -fdata-sections -Werror=switch -Wdocumentation -Wimplicit-fallthrough -Wunreachable-code -Woverloaded-virtual -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -O3 -DNDEBUG  -target x86_64-unknown-linux-gnu -fuse-ld=gold -shared -Wl,-soname,libswiftDemangle.so -o lib/libswiftDemangle.so lib/SwiftDemangle/CMakeFiles/swiftDemangle.dir/SwiftDemangle.cpp.o lib/SwiftDemangle/CMakeFiles/swiftDemangle.dir/MangleHack.cpp.o -L/tmp/swift-20230110-2076873-jfvu2m/build/llvm-linux-x86_64/./lib -Wl,-rpath,"\$ORIGIN:\$ORIGIN/../lib:\$ORIGIN/../lib/swift/linux"  lib/libswiftDemangling.a && :
> /home/linuxbrew/.linuxbrew/opt/binutils/bin/ld.gold: error: cannot open /home/linuxbrew/.linuxbrew/lib/libgcc_s.so.1: No such file or directory
> /home/linuxbrew/.linuxbrew/opt/binutils/bin/ld.gold: error: cannot open /home/linuxbrew/.linuxbrew/lib/libgcc_s.so.1: No such file or directory
> clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
> ```

Looks like this might be caused by something else being broken here? That is not swift being broken but binutils being broken which is quite major.

**Comment by cho-m** [comment]
> I can increase the disk space of the runner. Surprised it hit it though? (currently 150GB)

- ~36G zip
- Not much difference unpacked so +36G
- The logic copies files, so another +36G

So 108G.

Some large bottles like `texlive` (1.7G each) and `freeling` (1.1G each).

Maybe we can retry once with larger disk space runner. Alternatively, we can try either `pr-pull` or a manual upload like previous `icu4c` PR. Only risk is we sometimes hit permissions issues.

**Comment by Bo98** [comment]
> So 108G. I believe we allocate 100G - https://github.com/Homebrew/actions/blob/master/create-gcloud-instance/create-gcloud-instance.sh#L11

We no longer use this runner for publishing - we use GitHub's larger runners but that will have some overhead on that 150GB. I'll increase it to 300GB.

**Comment by cho-m** [comment]
> > So 108G. I believe we allocate 100G - https://github.com/Homebrew/actions/blob/master/create-gcloud-instance/create-gcloud-instance.sh#L11
> 
> We no longer use this runner for publishing - we use GitHub's larger runners but that will have some overhead on that 150GB. I'll increase it to 300GB.

Interesting. Not sure on size then unless rest is occupied by OS and software.

**Review by cho-m — APPROVED** [review]

**Comment by chenrui333** [comment]
@cho-m the disk space got increased, I think we can have another run.

**Comment by cho-m** [comment]
> > @Bo98 `swift` failure on Linux:
> > ```
> > FAILED: lib/libswiftDemangle.so 
> > : && /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/linux/super/llvm_clang++ -fPIC -Wno-unknown-warning-option -Werror=unguarded-availability-new -fno-stack-protector -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-class-memaccess -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -fdiagnostics-color -ffunction-sections -fdata-sections -Werror=switch -Wdocumentation -Wimplicit-fallthrough -Wunreachable-code -Woverloaded-virtual -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -O3 -DNDEBUG  -target x86_64-unknown-linux-gnu -fuse-ld=gold -shared -Wl,-soname,libswiftDemangle.so -o lib/libswiftDemangle.so lib/SwiftDemangle/CMakeFiles/swiftDemangle.dir/SwiftDemangle.cpp.o lib/SwiftDemangle/CMakeFiles/swiftDemangle.dir/MangleHack.cpp.o -L/tmp/swift-20230110-2076873-jfvu2m/build/llvm-linux-x86_64/./lib -Wl,-rpath,"\$ORIGIN:\$ORIGIN/../lib:\$ORIGIN/../lib/swift/linux"  lib/libswiftDemangling.a && :
> > /home/linuxbrew/.linuxbrew/opt/binutils/bin/ld.gold: error: cannot open /home/linuxbrew/.linuxbrew/lib/libgcc_s.so.1: No such file or directory
> > /home/linuxbrew/.linuxbrew/opt/binutils/bin/ld.gold: error: cannot open /home/linuxbrew/.linuxbrew/lib/libgcc_s.so.1: No such file or directory
> > clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
> > ```
> 
> Looks like this might be caused by something else being broken here? That is not swift being broken but binutils being broken which is a major issue if so.

As note, I was able to restore Linux `swift` bottle afterward. Failure seems to happen when `gcc` is installed and build tools (`llvm -> binutils (gold)`) are now using it.

Also mentioned issue in Linux Slack channel if anyone wants to discuss over there as this PR is now done.

**Pull request closed without merging**
